python/pyiceberg/catalog/__init__.py

```diff
@@ -25,6 +26,41 @@
 from pyiceberg.table.partitioning import UNPARTITIONED_PARTITION_SPEC, PartitionSpec
 from pyiceberg.table.sorting import UNSORTED_SORT_ORDER, SortOrder
 from pyiceberg.typedef import EMPTY_DICT, Identifier, Properties
+from pyiceberg.utils.config import Config, merge_config
+
+logger = logging.getLogger(__name__)
+
+_env_config = Config()
+
+
+def load_catalog(name: str, uri: str, **properties: str | None) -> Catalog:
+    from pyiceberg.catalog.rest import RestCatalog
+
+    supported_catalogs: dict[str, type[Catalog]] = {"http": RestCatalog}
+
+    try:
+        # In case Thrift isn't installed
+        from pyiceberg.catalog.hive import HiveCatalog
+
+        supported_catalogs["thrift"] = HiveCatalog
+    except ImportError:
+        logger.warning("Apache Hive not supported, to enable: pip install 'pyiceberg[hive]'")
+
+    if name:
+        env = _env_config.get_catalog_config(name)
+        conf = merge_config(env or {}, properties)
```

Comment:
Since `uri` was passed separately, I think we need to add URI to `conf` or else it won't be passed.

Reply:
I merged the `uri` into the `properties`